cmake/module/GenerateHeaders.cmake

```diff
@@ -5,17 +5,17 @@
 function(generate_header_from_json json_source_relpath)
   add_custom_command(
     OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${json_source_relpath}.h
-    COMMAND ${CMAKE_COMMAND} -DJSON_SOURCE_PATH=${CMAKE_CURRENT_SOURCE_DIR}/${json_source_relpath} -DHEADER_PATH=${CMAKE_CURRENT_BINARY_DIR}/${json_source_relpath}.h -P ${CMAKE_SOURCE_DIR}/cmake/script/GenerateHeaderFromJson.cmake
-    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${json_source_relpath} ${CMAKE_SOURCE_DIR}/cmake/script/GenerateHeaderFromJson.cmake
+    COMMAND ${CMAKE_COMMAND} -DJSON_SOURCE_PATH=${CMAKE_CURRENT_SOURCE_DIR}/${json_source_relpath} -DHEADER_PATH=${CMAKE_CURRENT_BINARY_DIR}/${json_source_relpath}.h -P ${PROJECT_SOURCE_DIR}/cmake/script/GenerateHeaderFromJson.cmake
```

Comment:
do you want to keep the CMAKE_CURRENT_BINARY_DIR/CMAKE_CURRENT_SOURCE_DIR here because we're in a submodule?

Reply:
I'm not sure if I understand your question. The `CMAKE_CURRENT_BINARY_DIR` and `CMAKE_CURRENT_SOURCE_DIR` variables are used as they are supposed to be used based on their semantics.

Reply:
K, thanks for checking, so it's `CMAKE_CURRENT_SOURCE_DIR` instead of `CMAKE_SOURCE_DIR` since we're in a child of a parent module, right? Wasn't a blocker, just want to make sure I understand the structure

Reply:
This code is an implementation of the `generate_header_from_json` function. Its parameter is a path to a file relative to the directory where the function is invoked. To construct the absolute path, we prepend that parameter with the `CMAKE_CURRENT_SOURCE_DIR` variable.

Reply:
Thank you for the context